src/lib/runner.rs

```diff
@@ -465,6 +466,11 @@ pub(crate) fn run_task_with_options(
                 };
             }
         } else {
+            // Restore env to state before expansion for condition evaluation
+            for (key, value) in env_before_current_task_env_expansion {
+                envmnt::set(key, value);
+            }
```

Comment:
the issue here is that new env vars are still stuck in the env. you only restore modified env vars.

Reply:
@wmmc88 rethinking this whole thing, can we just not do all this env changes and only expand with what we have?
it is unexpected bahvior that env changes before condition is met.
if the condition needs some env vars, let them be defined in a dependency task as is.
this is much more readable and expected

Reply:
I could rework the pr to not expand the env for the condition script, but I want to highlight that this means the some env will change before condition script and some will change after. Particularly, all the internal meta ends like `CARGO_MAKE_CURRENT_TASK*` will expand before condition, and the rest will expand after.

Reply:
i think what i'm saying is that for your PR, runner.rs does not need to change at all and control should be in the makefile author to setup env for conditions before hand

Reply:
Before condition evaluation in runner.rs, there still needs to be a small change to call a function to expand args for condition_script_runner_args. 

If I understand you correctly, the `CARGO_MAKE_CURRENT_TASK_*` env vars will expand to the updated values for the current task (which matches existing behavior), but other env vars will not be updated (i.e. I remove the env expansion).

Reply:
if its just to call expand ya thats ok.

env is updated for all vars. no env var gets special invocation

Reply:
made change to only explicitly expand the condition-script-runner-args